src/EnergyPlus/UnitarySystem.cc

```diff
@@ -13244,22 +13248,24 @@ namespace UnitarySystems {
                                                                                                       singleMode);
                             if ((state.dataLoopNodes->Node(OutletNode).HumRat - DesOutHumRat) < Acc) break;
                         }
+                        // make sure outlet HumRat is below set point before calling SolveRoot
+                        if (state.dataLoopNodes->Node(OutletNode).HumRat < DesOutHumRat) {
```

Comment:
Final revisions: so thinking about my other comment on the if test on break being identical to the if test here. If the if test on break is true it means the outlet condition was lower than the set point by Acc (definitely room to iterate). This if test only tests if the outlet condition is lower than the set point (i.e., could be so close that SolveRoot might fail?). Now I have to decide if this is correct or if they should be identical (i.e., which is more correct with less chance for failure). Probably safer to make them identical because we want to be sure SolveRoot can find a solution. If the outlet condition is closer to the set point than Acc then it is likely there is no need for iteration or a chance of SolveRoot failure (i.e., just use PLR=1). This of course is debatable.

Reply:
Now I am thinking that these should be flipped. break for loop if outlet <= set point (i.e., this speed can meet the set point), and iterate only if (outlet-set point) < Acc (i.e., no need to iterate if PLR would be very close to 1). If this speed can meet the set point, equally or just barely, then that is the speed to use. The next higher speed would be meeting the set point at cycling ratio = 0.  Kind of a brain teaser, and still debatable.

Reply:
I'm still trying to figure out the correct logic. This is cooling logic and Acc is positive at 0.001 for T and 0.000001 for w.

    Real64 constexpr Acc(1.e-3);       // Accuracy of solver result
    Real64 constexpr HumRatAcc(1.e-6); // Accuracy of solver result

So what the for loop break does is as the outlet node condition approaches the set point close enough, from above in cooling mode, it breaks out of the for loop. This means that the result of this speed, at PLR=1, is either very close above the set point or at or lower than the set point. I think this part is sound but welcome feedback.

    if ((state.dataLoopNodes->Node(OutletNode).HumRat - DesOutHumRat) < Acc) break

Now, when should we iterate? Certainly not if outlet node > set point because SolveRoot would fail with -2 (bounds of PLR = 0 to 1 do not enclose target). And also not if outlet node = set point because SolveRoot would also fail with -2 (because bounds should be on either side of target not one bound at the target) and we already have the correct answer anyway. So on the test for SolveRoot I think

    if ((outlet - set point) < -Acc) {
        SolveRoot();
    }

this means PLR will remain 1 until the result of the speed test is lower than the set point by Acc and we are sure a solution can be found. This also means that there is little chance of SolveRoot failure when the bounds are very close to the target. It also means that there is a little bit of a continuity issue with PLR where the outlet can float around the set point within Acc at PLR = 1 in certain instances. I can live with that to avoid future issues.

Open for debate and feedback.